csharp/GrpcOpenTracing/GrpcOpenTracing/Propagation/MetadataCarrier.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Grpc.Core;
+using OpenTracing.Propagation;
+
+namespace GrpcOpenTracing.Propagation
+{
+    internal class MetadataCarrier : ITextMap
+    {
+        private readonly Metadata metadata;
+        private readonly Dictionary<string, string> dictionary;
+
+        public MetadataCarrier(Metadata metadata)
+        {
+            this.metadata = metadata;
+            this.dictionary = metadata.Where(e => !e.IsBinary).ToDictionary(e => e.Key, e => e.Value);
```

Comment:
Might be more efficient to only return a converted object (or even better - a custom enumerator) in the GetEnumerator() method instead of having duplicate objects/logic everywhere.